cpp/src/copying/copy_range.cu

```diff
@@ -71,6 +83,65 @@ struct inplace_copy_range_dispatch {
   }
 };
 
+struct out_of_place_copy_range_dispatch {
+  cudf::column_view const& source;
+  cudf::column_view const& target;
+
+  template <typename T>
+  std::enable_if_t<cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>>
+  operator()(
+      cudf::size_type source_begin, cudf::size_type source_end,
+      cudf::size_type target_begin,
+      rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+      cudaStream_t stream = 0) {
+    auto p_ret = std::make_unique<cudf::column>(target, stream, mr);
+    if ((!p_ret->nullable()) && source.has_nulls()) {
+      p_ret->set_null_mask(
+        cudf::create_null_mask(p_ret->size(), cudf::ALL_VALID, stream, mr), 0);
+    }
+
+    if (source_end != source_begin) {  // otherwise no-op
+      auto ret_view = p_ret->mutable_view();
+      in_place_copy_range<T>(
+        source, ret_view, source_begin, source_end, target_begin, stream);
+    }
+
+    return p_ret;
+  }
+
+  template <typename T>
+  std::enable_if_t<not cudf::is_fixed_width<T>(), std::unique_ptr<cudf::column>>
```

Comment:
Strings are not the only fixed_width types (in the future). So I think this SFINAE should be specific to strings...

Reply:
Oops, I mistakenly fixed this for the in-place case :-(

I will revert the change for the in-place case and apply the change here.